projects/plugins/jetpack/modules/masterbar/admin-menu/class-dashboard-switcher-tracking.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Quick switcher tracking file.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Terms_Of_Service;
+use Automattic\Jetpack\Tracking;
+use Jetpack_Plan;
+
+/**
+ * Class Dashboard_Switcher_Tracking
+ */
+class Dashboard_Switcher_Tracking {
+
+	/**
+	 * Jetpack Tracking library will prefix the event name with "jetpack_*" automatically.
+	 */
+	const JETPACK_EVENT_NAME = 'dashboard_quick_switch_link_clicked';
+
+	const WPCOM_EVENT_NAME = 'wpcom_dashboard_quick_switch_link_clicked';
+	/**
+	 * Jetpack tracking object.
+	 *
+	 * @var Tracking
+	 */
+	private $tracking;
+
+	/**
+	 * Current site plan.
+	 *
+	 * @var string
+	 */
+	private $plan;
+
+	/**
+	 * The wpcom_tracks wrapper function.
+	 *
+	 * @var callable
+	 */
+	private $wpcom_tracking;
+
+	/**
+	 * Dashboard_Switcher_Tracking constructor.
+	 *
+	 * @param Tracking $tracking       Jetpack tracking object.
+	 * @param callable $wpcom_tracking A wrapper over wpcom event record.
+	 * @param string   $plan           The current site plan.
+	 */
+	public function __construct( Tracking $tracking, callable $wpcom_tracking, $plan ) {
```

Comment:
Instead of calling directly the static helper methods and initializing the Tracking class in this class, I've opted to inject them into the constructor because we decouple the logic that registers the event from these dependencies.

Besides this, it also allows us to easier test the implementation in our tests.